### PR TITLE
nirbhay/aws_s3_storage_lens

### PR DESCRIPTION
**Pull request by Nirbhay1997**
Signed-off-by: Nirbhay Kumar <nkumar@progress.com>

### Description

It contains all the singular and plural resources of the above resource. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_s3_storage_lenses.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    validate_parameters(required: %i(account_id))
+    @query_params = {}
+    raise ArgumentError, "#{@__resource_name__}: account_id must be provided" unless opts[:account_id] && !opts[:account_id].empty?
+    @query_params[:account_id] = opts[:account_id]
+    @table = fetch_data
+  end
+
+  def fetch_data
+    catch_aws_errors do
+      @table = @aws.rds_client.describe_db_proxy_endpoints(@query_params).map do |table|
```
Comment (soumyo13) [inline libraries/aws_s3_storage_lenses.rb:35]:
```suggestion
      @table = @aws.storage_client.list_storage_lens_configurations(@query_params).map do |table|
```

File: libraries/aws_s3_storage_lenses.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        table.db_proxy_endpoints.map { |table_name| {
+          id: table_name.id,
+          storage_lens_arn: table_name.storage_lens_arn,
+          home_region: table_name.home_region,
+          is_enabled: table_name.is_enabled,
+
```
Comment (soumyo13) [inline libraries/aws_s3_storage_lenses.rb:41]:
```suggestion
```

File: docs/resources/aws_s3_storage_lens.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+---
+title: About the aws_s3_storage_lens Resource
+platform: aws
+---
+
+# aws_rds_db_cluster_snapshot
```
Comment (soumyo13) [inline docs/resources/aws_s3_storage_lens.md:6]:
```suggestion
# aws_s3_storage_lens
```

File: docs/resources/aws_s3_storage_lens.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+Use `should` to check if the entity is available.
+
+    describe aws_s3_storage_lens(config_id: 'CONFIG_ID', account_id: 'ACCOUNT_ID') do
+      it { should be_available }
+    end
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `S3:Client:GetStorageLensConfigurationResult` action with `Effect` set to `Allow`.
```
Comment (soumyo13) [inline docs/resources/aws_s3_storage_lens.md:100]:
```suggestion
Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `S3Control:Client:GetStorageLensConfigurationResult` action with `Effect` set to `Allow`.
```

File: docs/resources/aws_s3_storage_lenses.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+Use `should` to check if the entity is available.
+
+    describe aws_s3_storage_lenses(config_id: 'CONFIG_ID') do
+      it { should be_available }
+    end
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `S3:Client:GetStorageLensConfigurationResult` action with `Effect` set to `Allow`.
```
Comment (soumyo13) [inline docs/resources/aws_s3_storage_lenses.md:79]:
```suggestion
Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `S3Control:Client:GetStorageLensConfigurationResult` action with `Effect` set to `Allow`.
```

File: docs/resources/aws_s3_storage_lenses.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
+---
+title: About the aws_s3_storage_lenses Resource
+platform: aws
+---
+
+# aws_rds_db_cluster_snapshot
```
Comment (soumyo13) [inline docs/resources/aws_s3_storage_lenses.md:6]:
```suggestion
# aws_s3_storage_lenses
```

File: libraries/aws_s3_storage_lens.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AWSS3StorageLens < AwsResourceBase
+  name 'aws_s3_storage_lens'
+  desc 'Retrieves information about a patch baseline.'
```
Comment (soumyo13) [inline libraries/aws_s3_storage_lens.rb:7]:
```suggestion
  desc 'Gets the Amazon S3 Storage Lens configuration.'
```

File: libraries/aws_s3_storage_lenses.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AWSS3StorageLenses < AwsResourceBase
+  name 'aws_s3_storage_lenses'
+  desc 'Returns information about S3 Storage lens.'
```
Comment (soumyo13) [inline libraries/aws_s3_storage_lenses.rb:7]:
```suggestion
  desc 'Gets a list of Amazon S3 Storage Lens configurations.'
```

**Review by sa-progress — APPROVED** [review]

**Comment by soumyo13** [comment]
This PR can't be merged as this PR is not tested.

**Pull request closed without merging**